plugins/modules/nd_pcv.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Cindy Zhao (@cizhao) <cizhao@cisco.com>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+import json
+import os
+import time
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: nd_pcv
+version_added: "0.2.0"
+short_description: Manage pre-change validation job
+description:
+- Manage pre-change validation job on Cisco Nexus Dashboard Insights (NDI).
+author:
+- Cindy Zhao (@cizhao)
+options:
+  insights_group:
+    description:
+    - The name of the insights group.
+    type: str
+    required: yes
+    aliases: [ fab_name, ig_name ]
+  name:
+    description:
+    - The name of the pre-change validation job.
+    type: str
+  description:
+    description:
+    - Description for the pre-change validation job.
+    type: str
+    aliases: [ descr ]
+  site_name:
+    description:
+    - Name of the Assurance Entity.
+    type: str
+    aliases: [ site ]
+  file:
+    description:
+    - Optional parameter if creating new pre-change analysis from file.
+    type: str
+  manual:
+    description:
+    - Optional parameter if creating new pre-change analysis from change-list (manual)
+    type: str
+  state:
+    description:
+    - Use C(query) for retrieving the version object.
+    type: str
+    choices: [ query ]
+    default: query
+extends_documentation_fragment: cisco.nd.modules
+'''
+
+EXAMPLES = r'''
+- name: Get prechange validation result
+  cisco.nd.nd_pcv:
+    insights_group: exampleIG
+    state: query
+  delegate_to: localhost
+  register: query_results
+- name: Get a specific prechange validation result
+  cisco.nd.nd_pcv:
+    insights_group: exampleIG
+    site_name: siteName
+    name: demoName
+    state: query
+  delegate_to: localhost
+  register: query_result
+- name: Create a new Pre-Change analysis from file
+  cisco.nd.nd_pcv:
+    insights_group: igName
+    site_name: siteName
+    name: demoName
+    file: configFilePath
+    state: present
+  delegate_to: localhost
+  register: present_pcv
+- name: Present Pre-Change analysis from manual changes
+  cisco.nd.nd_pcv:
+    insights_group: idName
+    site_name: SiteName
+    name: demoName
+    manual: |
+        [
+            {
+              "fvTenant": {
+                "attributes": {
+                  "name": "AnsibleTest",
+                  "dn": "uni/tn-AnsibleTest",
+                  "status": "deleted"
+                }
+              }
+            }
+        ]
+    state: present
+  delegate_to: localhost
+  register: present_pcv_manual
+- name: Delete Pre-Change analysis
+  cisco.nd.nd_pcv:
+    insights_group: igName
+    site_name: siteName
+    name: demoName
+    state: absent
+  delegate_to: localhost
+  register: rm_pcv
```

Comment:
probably no need for register on the rm one not sure if any of the others need to be registered except the query.

Reply:
removed register for deletion